core/ui/vault/chain/coin/metadata/CoinsMetadataManager.tsx

```
@@ -0,0 +1,54 @@
+      const coinInfo = coins.find(coin => areEqualCoins(coin, vaultCoin))
+
+      if (!coinInfo) {
+        return acc
+      }
+
+      const coinMetadata = pick(coinInfo, coinMetadataFields)
+      const vaultCoinMetadata = pick(vaultCoin, coinMetadataFields)
+
+      if (!areEqualRecords(coinMetadata, vaultCoinMetadata)) {
+        return [
+          ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)